### PR TITLE
Fix opensea attributes not showing up

### LINKED ISSUE #948: Should get all attributes from opensea into token instance
Android screenshot, should have the attributes in the last screenshot:
Screenshot 2019-10-22 21 15 15

iOS: has the opensea details in full but doesn't include the attributes from the tsml (bug reported here: https://github.com/AlphaWallet/alpha-wallet-ios/issues/1504)

Screenshot 2019-10-22 21 15 25

### PR DESCRIPTION
**Pull request by James-Sangalli**
fixes #948

### CODE REVIEW

File: app/src/main/java/com/alphawallet/app/service/AssetDefinitionService.java [hunk 1]
```diff
@@ -1203,6 +1238,11 @@ public StringBuilder getTokenAttrs(Token token, BigInteger tokenId, int count)
         TokenScriptResult.addPair(attrs, "tokenId", tokenId);
         TokenScriptResult.addPair(attrs, "ownerAddress", token.getWallet());
```
Comment (JamesSmartCell) [inline app/src/main/java/com/alphawallet/app/service/AssetDefinitionService.java]: This should be done with OO abstraction, also don't cast to ERC721 token just pass the token directly.

You could do something like this at the base level:

```
if (token instanceof ERC721Token) 
{
   addOpenSeaAttributes(attrs, token, tokenId);
}
```
Add the function getAssetForToken into class Token, return null, then specify it's an @Override in the ERC721Token definition.

File: app/src/main/java/com/alphawallet/app/service/OpenseaService.java [hunk 2]
```diff
@@ -78,7 +78,7 @@ public OpenseaService(Context ctx) {
                     foundTokens.put(asset.getAssetContract().getAddress(), token);
                 }
 
-                ((ERC721Token) token).tokenBalance.add(asset);
+                ((ERC721Token) token).addAssetToTokenBalanceAssets(asset);
```
Comment (JamesSmartCell) [inline app/src/main/java/com/alphawallet/app/service/OpenseaService.java]: This should be abstracted as above.

File: app/src/main/java/com/alphawallet/app/ui/widget/adapter/NonFungibleTokenAdapter.java [hunk 3]
```diff
@@ -135,7 +135,7 @@ protected void setERC721Tokens(Token token, String ticketId)
         int weight = 1; //use the same order we receive from OpenSea
 
         // populate the ERC721 items
-        for (Asset asset : ((ERC721Token)token).tokenBalance)
+        for (Asset asset : ((ERC721Token)token).getTokenAssets())
```
Comment (JamesSmartCell) [inline app/src/main/java/com/alphawallet/app/ui/widget/adapter/NonFungibleTokenAdapter.java]: Abstract

File: app/src/main/java/com/alphawallet/app/viewmodel/TransferTicketDetailViewModel.java [hunk 4]
```diff
@@ -252,7 +252,7 @@ public void openConfirm(Context ctx, String to, Token token, String tokenId, Str
     {
         //first find the asset within the token
         Asset asset = null;
-        for (Asset a : ((ERC721Token) token).tokenBalance)
+        for (Asset a : ((ERC721Token) token).getTokenAssets())
```
Comment (JamesSmartCell) [inline app/src/main/java/com/alphawallet/app/viewmodel/TransferTicketDetailViewModel.java]: Abstract

**Comment by James-Sangalli** [comment]
@JamesSmartCell see latest commit

**Pull request merged**